plugin.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	secretmanagerpb "google.golang.org/genproto/googleapis/cloud/secretmanager/v1"
+	yaml "gopkg.in/yaml.v2"
+)
+
+// Secret holds the parameters of the SecretProviderClass CRD. Links the GCP
+// secret resource name to a path in the filesystem.
+type Secret struct {
+	// The resource name of the SecretVersion in the format projects/*/secrets/*/versions/*.
+	ResourceName string `json:"resourceName" yaml:"resourceName"`
+	// The filename where the contents of the secret are to be written.
+	FileName string `json:"fileName" yaml:"fileName"`
+}
+
+// StringArray holds the 'secrets' key of the SecretProviderClass parameters.
```

Comment:
Why not have this be a `[]string` directly?

Reply:
I created #6 for this and would like to discuss there and reach out to the other plugins to see if there is a reason we should not do this.

Reply:
And the reason I can't do it _here_ is that I need some object to capture the `array` field name